src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -194,6 +195,51 @@ extension ExposureSubmissionCoordinator {
 		present(vc)
 	}
 
+	func showSymptomsScreen() {
+		let vc = createSymptomsViewController(
+			onPrimaryButtonTap: { [weak self] selectedSymptomsOption in
+				switch selectedSymptomsOption {
+				case .yes:
+					self?.showSymptomsOnsetScreen()
+				case .no:
+					self?.symptomsOnset = .nonSymptomatic
+					self?.showWarnOthersScreen()
+				case .preferNotToSay:
+					self?.symptomsOnset = .noInformation
+					self?.showWarnOthersScreen()
+				}
```

Comment:
I guess @melloskitten's comment was aimed at these changes here in the coordinator (same applies for the showSymptomsOnsetScreen()). I would propose to handle this logic in the respective view controllers (if possible? Is there a specific reason why it is done here?). I would try to keep the coordinator free of any logic besides routing and dependency management.

Reply:
I am currently working on a refactoring of this part outside of this story. I think the switch is actually mainly routing and belongs here. I will move the symptomsOnset property as well as the exposure submission service to a separate coordinator model to make it more testable. The previous implementation seemed more like a factory to me than a coordinator. I don't think the view controller itself should have any clue which view controller precedes it or comes after it, that's why in this first step I pulled that information out of there. Would be great if we can leave it here for this story and discuss it in my pull request that will follow in the next days :)

Reply:
thank you for clarifying! 👍  I understand your reasoning and I am looking forward to see your future PR that addresses your points.